beetsplug/replaygain.py

```diff
@@ -81,13 +86,21 @@ def __init__(self, config, log):
         self._log = log
 
     def compute_track_gain(self, items):
+        """Computes the track gain of the given tracks, returns a list
+        of Gain objects.
+        """
         raise NotImplementedError()
 
     def compute_album_gain(self, album):
-        # TODO: implement album gain in terms of track gain of the
```

Comment:
I have not restored this `TODO` after removing the wrong implementation as I do not think it is possible to implement a correct default implementation without creating a new item/file (a concatenation of all the album's items). E.g. when using the EBU R128 spec we don't know how many gating blocks in each track were over the gating threshold, and the album gain cannot be calculated from the track gains without that information.
One could use the length of the tracks as a proxy for the number of gating blocks. That might get reasonable results but would still be _wrong_.

Another option would be to implement a method computing the album gain from the track gains and weights (i.e. number of gating blocks above threshold) according to EBU R128. Thus, the main album gain calculation could be used by all backends that can provide some weights.
It should be really easy to move that calculation from `FffmpegBackend` to `Backend`, but I do not know if any other `Backend` could possibly make use of that?

Reply:
OK, thanks for carefully investigating. I'ma also not aware of any other backend that can provide the necessary information about weighing for tracks.